test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh.py

```diff
@@ -0,0 +1,285 @@
+        new_ops_name = [op.name() for op in dist_program.global_block().ops]
+
+        all_gather_ops = []
+        slice_ops = []
+        for i, op in enumerate(new_ops):
+            if op.name() == "pd_op.c_allgather":
+                all_gather_ops.append(op)
+            elif op.name() == "pd_op.slice":
+                slice_ops.append(op)
+        assert len(all_gather_ops) == 2
+        assert len(slice_ops) == 2
+        rank_id = dist.get_rank()
+
+        # check the first all_gather
+        op = all_gather_ops[0]
+        if rank_id == 0 or rank_id == 1:
+            process_ids = [0, 1]
+        elif rank_id == 2 or rank_id == 3:
+            process_ids = [2, 3]
+        tgt_operand = (
+            process_ids,
+            [-1, 0, -1],
+            {},
+        )  # process_ids, dims_mapping, partial_status
+        tgt_result = (process_ids, [-1, -1, -1], {})
+        tgt_in_value = (self._mesh.process_ids, [0, 1, -1], {})
+        tgt_out_value = (self._mesh.process_ids, [0, -1, -1], {})
+        self.validate(op, tgt_operand, tgt_result, tgt_in_value, tgt_out_value)
+
+        # check the second all_gather
+        op = all_gather_ops[1]
+        if rank_id == 0 or rank_id == 2:
+            process_ids = [0, 2]
+        elif rank_id == 1 or rank_id == 3:
+            process_ids = [1, 3]
+        tgt_operand = (
+            process_ids,
+            [0, -1, -1],
+            {},
+        )  # process_ids, dims_mapping, partial_status
+        tgt_result = (process_ids, [-1, -1, -1], {})
+        tgt_in_value = (self._mesh.process_ids, [0, -1, -1], {})
+        tgt_out_value = (self._mesh.process_ids, [-1, -1, -1], {})
+        self.validate(op, tgt_operand, tgt_result, tgt_in_value, tgt_out_value)
+
+        # check the first slice
+        op = slice_ops[0]
+        if rank_id == 0 or rank_id == 2:
+            process_ids = [0, 2]
+        elif rank_id == 1 or rank_id == 3:
+            process_ids = [1, 3]
+        tgt_operand = (process_ids, [-1, -1, -1], {})
+        tgt_result = (process_ids, [-1, 0, -1], {})
+        tgt_in_value = (self._mesh.process_ids, [-1, -1, -1], {})
+        tgt_out_value = (self._mesh.process_ids, [-1, 0, -1], {})
+        self.validate(op, tgt_operand, tgt_result, tgt_in_value, tgt_out_value)
+
+        # check the second slice
+        op = slice_ops[1]
+        if rank_id == 0 or rank_id == 1:
+            process_ids = [0, 1]
+        elif rank_id == 2 or rank_id == 3:
+            process_ids = [2, 3]
+        tgt_operand = (process_ids, [-1, -1, -1], {})
+        tgt_result = (process_ids, [0, -1, -1], {})
+        tgt_in_value = (self._mesh.process_ids, [-1, 0, -1], {})
+        tgt_out_value = (self._mesh.process_ids, [1, 0, -1], {})
+        self.validate(op, tgt_operand, tgt_result, tgt_in_value, tgt_out_value)
+
+    def run_test_cases(self):
+        self.run_pp_to_rr_case()
+        self.run_pr_to_rs_case()
+        self.run_pr_to_ss_case()
+        self.run_ss_to_ss_case()
+
```

Comment:
add test for  ss_to_rr, rs_to_sr (shard in same mesh dim and diff mesh dim) in future.